server/py/framework/utils/clients/messaging.py

```diff
@@ -0,0 +1,268 @@
+        if raise_on_failure:
+            mlrun.errors.raise_for_status(response)
+
+    @staticmethod
+    def _resolve_request_kwargs_from_request(
+        request: fastapi.Request = None, json: typing.Optional[dict] = None, **kwargs
+    ) -> dict:
+        request_kwargs = {}
+        if request:
+            json = json if json else {}
+            request_kwargs.update({"json": json})
+            request_kwargs.update({"headers": dict(request.headers)})
+            request_kwargs.update({"params": dict(request.query_params)})
+            request_kwargs.update({"cookies": request.cookies})
+            request_kwargs["headers"].setdefault(
+                "x-request-id", request.state.request_id
+            )
+
+        # mask clients host with worker's host
+        origin_host = request_kwargs.get("headers", {}).pop("host", None)
+        if origin_host:
+            # original host requested by client
+            request_kwargs["headers"]["x-forwarded-host"] = origin_host
+
+        # let the http client calculate it itself
+        # or we will hit serious issues with reverse-proxying (client<->worker<->chief) requests
+        request_kwargs.get("headers", {}).pop("content-length", None)
+
+        for cookie_name in list(request_kwargs.get("cookies", {}).keys()):
+            # defensive programming - to avoid setting reserved cookie names and explode
+            # e.g.: when setting "domain" cookie, it will explode, see python internal http client for more details.
+            if http.cookies.Morsel().isReservedKey(cookie_name):
+                del request_kwargs["cookies"][cookie_name]
+
+            # iguazio auth cookies might include special characters. to ensure the http client wont escape them
+            # we will url-encode them (aka quote), so the value would be safe against such escaping.
+            # e.g.: instead of having "x":"y" being escaped to "\"x\":\"y\"", it will be escaped to "%22x%22:%22y%22"
+            elif cookie_name == "session" and mlrun.mlconf.is_running_on_iguazio():
+                # unquote first, to avoid double quoting ourselves, in case the cookie is already quoted
+                unquoted_session = urllib.parse.unquote(
+                    request_kwargs["cookies"][cookie_name]
+                )
+                request_kwargs["cookies"][cookie_name] = urllib.parse.quote(
+                    unquoted_session
+                )
+
+        request_kwargs.update(**kwargs)
+        return request_kwargs
+
+
+class Client(BaseClient):
+    def __init__(self) -> None:
+        super().__init__()
+        # Retry session is for internal messaging
+        self._retry_session: typing.Optional[mlrun.utils.AsyncClientWithRetry] = None
+        self._discovery = framework.utils.clients.discovery.Client()
+
+    def post(self):
+        pass
+
+    def put(self):
+        pass
+
+    def delete(self):
+        pass
+
+    def get(self):
+        pass
```

Comment:
why not async?

Reply:
I was contemplating if there is any point to keep these in this PR. Decided to remove them